naga/src/lib.rs

```diff
@@ -2186,6 +2186,8 @@ pub struct EntryPoint {
     pub early_depth_test: Option<EarlyDepthTest>,
     /// Workgroup size for compute stages
     pub workgroup_size: [u32; 3],
+    /// Override expressions for workgroup size
+    pub workgroup_size_overrides: Option<[Option<Handle<Override>>; 3]>,
```

Comment:
I think this can be a `Handle<Expression>` but please document that it points into `global_expressions`.

Reply:
We should also check in the validator that the `Handle` is valid. Example:

https://github.com/gfx-rs/wgpu/blob/ed2940d8697f3eb136536f1a12f4602df996e2ee/naga/src/valid/handles.rs#L90-L99

Reply:
Since the `Handle<Expression>`s point into the `global_expressions` arena, isn't it validated by `validate_const_expression_handles` just above where you quoted?
https://github.com/gfx-rs/wgpu/blob/ed2940d8697f3eb136536f1a12f4602df996e2ee/naga/src/valid/handles.rs#L75-L77

Reply:
Oh, I see, that doesn't cover the case where the handle isn't in the arena

Reply:
wait, you said it's a `Handle<Expression>` indexing into the `global_expressions` arena, but it's actually a `Handle<Override>` indexing into the overrides arena, which is unambiguous based on the type. Other than that, I think I've fulfilled the request to validate the handles.

Reply:
or do you mean it *should* be `Handle<Expression>`?

Reply:
Yeah, I meant that there didn't seem to be a need to insert an Override in the overrides Arena and we could directly use the `Handle<Expression>`. They probably got removed due to the compaction phase. We need to mark the expressions as used. Example:

https://github.com/gfx-rs/wgpu/blob/b876b8c2816441bd53f4d787739db929f9da4dd4/naga/src/compact/mod.rs#L58-L64